### PR TITLE
docs: add docker setup doc for dotnet

### PR DESCRIPTION
**Pull request by Harish-osmosys**
Created a docker setup guidline for dotnet project 

## Summary by CodeRabbit

- **Documentation**
	- Updated the guide on dockerizing .NET Core applications to include setup, common issues, best practices, and deployment instructions.

### CODE REVIEW

File: docker-setups/dotnet-github.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
+# Dockerizing .NET project
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docker-setups/dotnet-github.md]: Updated in https://github.com/OsmosysSoftware/dev-standards/pull/54/commits/ab2acdf0105a5b3b58122788df56bc700bc99d7e

File: docker-setups/dotnet-github.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+      - [Push Dockerized application to DockerHub](#push-dockerized-application-to-dockerhub)
+      - [Run Dockerized application](#run-dockerized-application)
+  - [4. Addressing Common Issues](#4-addressing-common-issues)
+  - [5. Best Practices for Enhanced Dockerization](#5-best-practices-for-enhanced-dockerization)
+  - [6. Conclusion](#6-conclusion)
+## 1. Introduction
+
+### Purpose
+
+The purpose of this document is to provide a step-by-step guide for dockerizing an .NET Core application. Dockerizing ensures consistent dependency versions and effortless cross-environment compatibility, simplifies development and deployment processes significantly.
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docker-setups/dotnet-github.md]: Updated in https://github.com/OsmosysSoftware/dev-standards/pull/54/commits/ab2acdf0105a5b3b58122788df56bc700bc99d7e

File: docker-setups/dotnet-github.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+### Scope
+
+The scope of this document covers the following
+
+1. Docker setup for an .NET Core application
+2. How to Deploy an Dockerized application using DockerHub
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docker-setups/dotnet-github.md]: Updated in https://github.com/OsmosysSoftware/dev-standards/pull/54/commits/ab2acdf0105a5b3b58122788df56bc700bc99d7e

File: docker-setups/dotnet-github.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+[Back to top](#table-of-contents)
+
+## 2. Prerequisites
+
+Before setting up the dockerize the application, ensure you have the following prerequisites:
+
+- A GitHub account with access to your target repository.
+- An Dotnet application repository hosted on GitHub.
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docker-setups/dotnet-github.md]: Updated in https://github.com/OsmosysSoftware/dev-standards/pull/54/commits/ab2acdf0105a5b3b58122788df56bc700bc99d7e

File: docker-setups/dotnet-github.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+- **Persistent Data with Volumes:** Mount volumes using the `-v` flag with `docker run` to persist application data outside of the container, ensuring data survives container restarts.
+- **Environment Variables:** Manage configuration details within your container by defining environment variables in your Dockerfile or during runtime using -e with docker run.
+
+[Back to top](#table-of-contents)
+
+## 6. Conclusion
+
+### Benefits of Docker
+
+Setting up a workflow for your Angular application offers several benefits:
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docker-setups/dotnet-github.md]: Updated in https://github.com/OsmosysSoftware/dev-standards/pull/54/commits/ab2acdf0105a5b3b58122788df56bc700bc99d7e

File: docker-setups/dotnet-github.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+5. Access your .NET application by visiting http://your-server-ip:8080 in a web browser.
+
+## 4. Addressing Common Issues:
+
+- **Missing Dependencies:** Ensure all necessary dependencies (NuGet packages, libraries) are included in your Dockerfile or copied into the container image. Utilize `docker logs` to inspect container logs for dependency-related errors.
+- **Port Conflicts:** Verify that the exposed port in your Dockerfile doesn't clash with another container or application running on your host machine. Use `docker ps` to list running containers and their ports.
+- **Application Errors:** If your application encounters errors inside the container, use `docker logs` to view container logs and pinpoint the root cause. You might need to adjust your Dockerfile commands or application configuration.
+- **Permissions Issues:** Double-check that your application has the required permissions to access files or resources within the container. Use `docker exec` to enter the container and examine file permissions.
+
+## 5.Best Practices for Enhanced Dockerization:
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docker-setups/dotnet-github.md]: Updated in https://github.com/OsmosysSoftware/dev-standards/pull/54/commits/ab2acdf0105a5b3b58122788df56bc700bc99d7e

**Review by sameer-s-b — APPROVED** [review]

**Pull request merged**